packages/connect-disputable-voting/src/models/DisputableVoting.ts

```diff
@@ -119,4 +153,20 @@ export default class DisputableVoting {
       this.#connector.onVoter(this.voterId(voterAddress), callback)
     )
   }
+
+  async newVote(script: string, context: string, signerAddress: string): Promise<ForwardingPath> {
+    const intent = await this.#app.intent('newVote', [script, utils.toUtf8Bytes(context)], { actAs: signerAddress })
+
+    // approve action collateral
+    const agreement = await this.agreement()
```

Comment:
I wonder if we can subscribe to the agreement data and cache that information in the app connector. My concern is the amount of request to the subgraph, do you think is relevant to consider it in this case?